frontend/src/components/GoBack/GoBack.js

```diff
@@ -0,0 +1,32 @@
+import "./GoBack.scss";
+import React from "react";
+import { IconButton } from "@material-ui/core";
+import ConfirmationDialog from "../ConfirmationDialog/ConfirmationDialog";
+
+const defaultBack = (
+  <IconButton onClick={() => window.history.back()}>
+    <i className="fas fa-arrow-left" />
+  </IconButton>
+);
+
+const handleSuccess = () => {
+  window.history.back();
+};
+
+const gameBack = (
+  <div>
+    <ConfirmationDialog title="Warning!" success={handleSuccess}>
+      Are you sure you want to leave the game?
+    </ConfirmationDialog>
+  </div>
+);
+
+const GoBack = () => {
+  return (
+    <div className="GoBack">
+      {window.location.pathname === "/game" ? gameBack : defaultBack}
```

Comment:
I think this is somewhat of an anti-pattern. How I think the code would look cleaner:

Create one button and declare it here. Give it a handler that you declare outside. Do not write any conditional on this line. Instead, check for the route path inside the function handler, and either open the dialog or just simply go back in history (Also, check for router history, not `window.location`). This will also require that you delete the button part from your `ConfirmationDialog`, and that you pass a prop to the `ConfirmationDialog` saying if it should be open or not (handled by your function handler).

So in the end you have:
* 1 button
* 1 dialog
* 1 handler function which checks for router history and either directly goes back or opens the dialog
* 1 state variable that controls whether the dialog is open or not

This will also make a lot more sense for the `keepMounted` prop that you passed to the dialog since then it will always be mounted in comparison to conditional mounting that is right now.

If it is not clear what I meant please ask and I will explain further.